### ISSUE TITLE
RuntimeError in "test_benchmark_ddpg"

### ISSUE THREAD

**Issue by FarnazAdib** [rlworkgroup/garage#580]
Hi.

I am trying to run  _~/tests/benchmarks.test_benchmark_ddpg.py_.

Each task will be run _trial_ times, which is set to 3. When the first trial is finished, I receive the error below. It seems that the graph is finalized and it is not possible to modify it in the next trial. 

I would be thankful if you let me know how I can solve the problem?
Thank you,

> Logging to ./benchmark_ddpg/2019-03-11-09-05-00-204088/InvertedDoublePendulum-v2/trail_2_seed_65/baselines
> rank 0: seed=65, logdir=./benchmark_ddpg/2019-03-11-09-05-00-204088/InvertedDoublePendulum-v2/trail_2_seed_65/baselines
> scaling actions by [1.] before executing in env
> /home/farnaz/miniconda2/envs/garage/lib/python3.6/site-packages/baselines/logger.py:372: ResourceWarning: unclosed file <_io.TextIOWrapper name='./benchmark_ddpg/2019-03-11-09-05-00-204088/InvertedDoublePendulum-v2/trail_1_seed_78/baselines/progress.csv' mode='w+t' encoding='UTF-8'>
>   Logger.CURRENT = Logger(dir=dir, output_formats=output_formats)
> /home/farnaz/miniconda2/envs/garage/lib/python3.6/site-packages/baselines/logger.py:372: ResourceWarning: unclosed file <_io.TextIOWrapper name='./benchmark_ddpg/2019-03-11-09-05-00-204088/InvertedDoublePendulum-v2/trail_1_seed_78/baselines/log.txt' mode='wt' encoding='UTF-8'>
>   Logger.CURRENT = Logger(dir=dir, output_formats=output_formats)
> Traceback (most recent call last):
> 
>   File "<ipython-input-2-f9a11b3f09b8>", line 1, in <module>
>     runfile('/home/farnaz/Documents/Packages/garage/FarnazTest/test_Bench', wdir='/home/farnaz/Documents/Packages/garage/FarnazTest')
> 
>   File "/home/farnaz/miniconda2/envs/garage/lib/python3.6/site-packages/spyder_kernels/customize/spydercustomize.py", line 786, in runfile
>     execfile(filename, namespace)
> 
>   File "/home/farnaz/miniconda2/envs/garage/lib/python3.6/site-packages/spyder_kernels/customize/spydercustomize.py", line 110, in execfile
>     exec(compile(f.read(), filename, 'exec'), namespace)
> 
>   File "/home/farnaz/Documents/Packages/garage/FarnazTest/test_Bench", line 11, in <module>
>     TestBenchmarkDDPG.test_benchmark_ddpg(unittest.TestCase)
> 
>   File "/home/farnaz/Documents/Packages/garage/tests/benchmarks/test_benchmark_ddpg.py", line 94, in test_benchmark_ddpg
>     baselines_csv = run_baselines(env, seed, baselines_dir)
> 
>   File "/home/farnaz/Documents/Packages/garage/tests/benchmarks/test_benchmark_ddpg.py", line 244, in run_baselines
>     batch_size=64)
> 
>   File "/home/farnaz/miniconda2/envs/garage/lib/python3.6/site-packages/baselines/ddpg/training.py", line 28, in train
>     reward_scale=reward_scale)
> 
>   File "/home/farnaz/miniconda2/envs/garage/lib/python3.6/site-packages/baselines/ddpg/ddpg.py", line 70, in __init__
>     self.obs0 = tf.placeholder(tf.float32, shape=(None,) + observation_shape, name='obs0')
> 
>   File "/home/farnaz/miniconda2/envs/garage/lib/python3.6/site-packages/tensorflow/python/ops/array_ops.py", line 1747, in placeholder
>     return gen_array_ops.placeholder(dtype=dtype, shape=shape, name=name)
> 
>   File "/home/farnaz/miniconda2/envs/garage/lib/python3.6/site-packages/tensorflow/python/ops/gen_array_ops.py", line 5206, in placeholder
>     "Placeholder", dtype=dtype, shape=shape, name=name)
> 
>   File "/home/farnaz/miniconda2/envs/garage/lib/python3.6/site-packages/tensorflow/python/framework/op_def_library.py", line 787, in _apply_op_helper
>     op_def=op_def)
> 
>   File "/home/farnaz/miniconda2/envs/garage/lib/python3.6/site-packages/tensorflow/python/util/deprecation.py", line 488, in new_func
>     return func(*args, **kwargs)
> 
>   File "/home/farnaz/miniconda2/envs/garage/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 3246, in create_op
>     self._check_not_finalized()
> 
>   File "/home/farnaz/miniconda2/envs/garage/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 2919, in _check_not_finalized
>     raise RuntimeError("Graph is finalized and cannot be modified.")
> 
> RuntimeError: Graph is finalized and cannot be modified.

**Comment by ryanjulian** [comment]
Thanks for the report. @CatherineSue Can help you with this one.

I should add that we generally don't run the benchmark tests during normal CI testing, since they take a very long time to complete. We run them on a fixed schedule on a different machine in the background. We mark the benchmark tests with the attribute `huge` and then filter them from the test suite (see command below).

If you want to run the CI test suite, you can either use the Makefile (recommended, requires Docker+docker-compose):
```sh
make test
```
 or you can run the tests directly on your machine (no Docker):
```sh
nose2 -c setup.cfg -v --with-id -E 'not cron_job and not huge and not flaky'
```

**Comment by CatherineSue** [comment]
Hi @FarnazAdib, thanks for reporting. This issue raises because in the second trial, baselines tries to use the `tf.Graph` created in the last trial, which is finalized and can't be modified.

This problem is addressed in #579.

**Comment by ryanjulian** [comment]
@CatherineSue has merge a fix. Please reopen if you still have this issue with the latest `master`
